### PR TITLE
React i18next

### PR DESCRIPTION
**Pull request by pettinarip**
## Description

Implement `react-i18next` by using the next plugin https://github.com/microapps/gatsby-plugin-react-i18next

- GC preview: https://ethereumorgwebsitedev01-reacti18next.gatsbyjs.io/
- Netlify preview: [https://pettinarip-eth-org.netlify.app/](https://pettinarip-eth-org.netlify.app/en/)

Known issues:

- Custom 404 pages are not working on the GC env. Still under investigation but I don’t think it is a blocker. Its working fine on Netlify env.

## Related Issue

#8345

### CODE REVIEW

File: gatsby-browser.tsx [hunk 1]
```diff
@@ -16,53 +12,6 @@ import "@formatjs/intl-locale/polyfill"
...
-    const queryParams = search || ""
-    const newUrl = withPrefix(`/${detected}${originalPath}${queryParams}`)
-    window.localStorage.setItem("eth-org-language", detected)
-    window.location.replace(newUrl)
-
-    return null
-  }
-
-  return <Layout {...props}>{element}</Layout>
-}
```
Comment (pettinarip) [inline gatsby-browser.tsx:68]: `gatsby-browser` & `gatsby-ssr` content was replaced by the usage of `gatsby-plugin-layout`.

File: src/components/Staking/StakingHierarchy.tsx [hunk 2]
```diff
@@ -287,9 +290,11 @@ const StakingHierarchy: React.FC<IProps> = () => {
...
         </Glyph>
         <Content>
-          <p>
-            <Translation id="page-staking-hierarchy-solo-p1" />
-          </p>
+          <p
+            dangerouslySetInnerHTML={{
+              __html: t("page-staking-hierarchy-solo-p1"),
+            }}
+          />
```
Comment (pettinarip) [inline src/components/Staking/StakingHierarchy.tsx:297]: This is a temporary implementation. In the next iteration we are planning in using the following approach: https://react.i18next.com/latest/trans-component#using-with-react-components where we will also need to modify our translation files.

File: src/pages-conditional/wallets.tsx [hunk 3]
```diff
@@ -494,7 +478,16 @@ export const listImage = graphql`
 `
 
 export const query = graphql`
-  query WalletsPage {
+  query WalletsPage($language: String!) {
+    locales: allLocale(filter: { language: { eq: $language } }) {
```
Comment (pettinarip) [inline src/pages-conditional/wallets.tsx:482]: On each page, we need to fetch the selected language strings.

**Comment by minimalsm** [comment]
When reviewing I noticed a bug with our link styling for links embedded as HTML in JSON files - #8547.

**Comment by wackerow** [comment]
Noticed our intl tutorials aren't loading properly
image

**Comment by pettinarip** [comment]
> Noticed our intl tutorials aren't loading properly

Uhmm good catch @wackerow! I'm looking into it now. Thanks

**Comment by minimalsm** [comment]
Noticed this on the other PR, but it's also happening here.

If a string is not available in a language, I would expect it to fall back to English strings. This is what happens in this PR:

Screenshot 2022-11-17 at 19 01 16

[English](https://ethereumorgwebsitereacti18n-i18nparsehtmljson62891.gtsb.io/en/layer-2)

[Chinese - A language with translated JSON strings](https://ethereumorgwebsitereacti18n-i18nparsehtmljson62891.gtsb.io/zh/layer-2)

[Slovenian - A language **without** translated JSON files](https://ethereumorgwebsitereacti18n-i18nparsehtmljson62891.gtsb.io/sl/layer-2)

Screenshot 2022-11-17 at 19 01 16

**Comment by pettinarip** [comment]
> Noticed this on the other PR, but it's also happening here.
> 
> If a string is not available in a language, I would expect it to fall back to English strings. This is what happens in this PR:

Good catch @minimalsm I thought I had that covered! will take a look, thanks!

**Review by corwintines — APPROVED** [review]
LGTM, nice one

Noting, we need to update [/docs/best-practices.md](https://github.com/ethereum/ethereum-org-website/blob/dev/docs/best-practices.md#%EF%B8%8F-translation-initiative)

**Review by minimalsm — APPROVED** [review]
LGTM :)

**Comment by pettinarip** [comment]
@corwintines I'll create a new PR with those updates 👍🏼 thanks for pointing that out.

**Pull request merged**